connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorCheckpointConfig.java

```diff
@@ -166,6 +167,30 @@ Duration consumerPollTimeout() {
         return Duration.ofMillis(getLong(CONSUMER_POLL_TIMEOUT_MILLIS));
     }
 
+    public static Map<String, String> validate(Map<String, String> configs) {
+        Map<String, String> invalidConfigs = new HashMap<>();
+
+        // No point to validate when connector is disabled.
+        if ("false".equals(configs.getOrDefault(ENABLED, "true"))) {
+            return invalidConfigs;
+        }
+
+        boolean emitCheckpointDisabled = "false".equals(configs.getOrDefault(EMIT_CHECKPOINTS_ENABLED, "true"));
+        boolean syncGroupOffsetsDisabled = "false".equals(configs.getOrDefault(SYNC_GROUP_OFFSETS_ENABLED, "true"));
+
+        if (emitCheckpointDisabled && syncGroupOffsetsDisabled) {
+            Arrays.asList(SYNC_GROUP_OFFSETS_ENABLED, EMIT_CHECKPOINTS_ENABLED).forEach(configName -> {
+                invalidConfigs.putIfAbsent(configName, "MirrorCheckpointConnector can't run with both " + SYNC_GROUP_OFFSETS_ENABLED + " and " +
```

Comment:
It seems to me `EMIT_CHECKPOINTS_ENABLED` does not obstruct `MirrorCheckpointConnector` from running since it is used to update consumer groups offsets of target cluster. By contrast, `SYNC_GROUP_OFFSETS_ENABLED` do impact the `MirrorCheckpointConnector`


https://github.com/apache/kafka/blob/trunk/connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorCheckpointConnector.java#L121

Reply:
My understanding that it will not start a task so the connector would be doing nothing without any feedback or indication for why. If this is problematic I would suggest remove this check as this was a suggestion out of the scope of the KIP to begin with. @C0urante and @chia7712 WDYT?

Reply:
I didn't realize that we disabled the connector from generating task configs if checkpoints were disabled. In that case, I think we should remove the check for `sync.group.offsets.enabled` altogether and just check `emit.checkpoints.enabled`. But generally I agree with @OmniaGM that if we don't generate task configs for the connector when it's configured that way, it's best to surface this as a validation error instead of silently degrading to a no-op.

Reply:
> It seems to me EMIT_CHECKPOINTS_ENABLED does not obstruct MirrorCheckpointConnector from running since it is used to update consumer groups offsets of target cluster. By contrast, SYNC_GROUP_OFFSETS_ENABLED do impact the MirrorCheckpointConnector

I just notice that I use the incorrect config name :(

> My understanding that it will not start a task so the connector would be doing nothing without any feedback or indication for why.

agree and my point was "we should return invalid config if `EMIT_CHECKPOINTS_ENABLED=false`", because `SYNC_GROUP_OFFSETS_ENABLED=false` is allowed to `MirrorCheckpointConnector` . For example: in this case: `EMIT_CHECKPOINTS_ENABLED=false` and `SYNC_GROUP_OFFSETS_ENABLED=true` we should return invalid configs, right?

Reply:
I removed the validation for `sync.group.offsets.enabled` and kept `emit.checkpoints.enabled` validation only